include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -271,8 +300,97 @@ namespace RINGMesh
 
         GEO::AttributesManager attributes_manager_;
     };
+    ALIAS_2D_AND_3D( CartesianGridBase );
+
+    template < index_t DIMENSION >
+    class CartesianGrid final : public CartesianGridBase< DIMENSION >
+    {
+        friend class CartesianGridBuilder< DIMENSION >;
+    };
     ALIAS_2D_AND_3D( CartesianGrid );
 
+    template <>
+    class CartesianGrid< 3 > final : public CartesianGridBase< 3 >
+    {
+        friend class CartesianGridBuilder< 3 >;
+
+    public:
+        CartesianGrid( ivec3 nb_cells_in_each_direction,
+            ReferenceFrame3D vec_cartesian_axis )
+            : CartesianGridBase(
+                  nb_cells_in_each_direction, vec_cartesian_axis ),
+              grid_cage_()
+        {
+            for( auto i : range( 3 ) )
+            {
+                vec3 plane_point{ cartesian_frame_.origin() };
+                vec3 plane_normal{ cartesian_frame_[i] };
+                grid_cage_.emplace_back(
```

Comment:
Why not grid_cage_.emplace_back( plane_normal, plane_point ); I di not like all this type redondancy.

Reply:
Will it create a Geometry::Plane and place it in the vector if I write this ?

Reply:
Exactly my point! You are talking about two steps: 1) new Plane, 2) add it to the vector. Using what I told you means directly create the Plane inside the vector.